### PR TITLE
Refactoring current code database

### PR DESCRIPTION
**Pull request by cepa995**
- Changes to dicom.py
- Changes to requiretements.txt
- Added preprocessing.py
- Added visualization.py

### CODE REVIEW

File: toolbox/dicom.py [hunk 1]
```diff
@@ -97,43 +107,75 @@ def get_elements(self, ds):
...
-    @classmethod
-    def get_df(cls, df_dcm_file):
-        dicom_tag_processor = DicomTagProcessor()
-        subject_list = []
-        for _, row in df_dcm_file.iterrows():
-            ds = pydicom.dcmread(row["path"], stop_before_pixels=True)
-            elements = dicom_tag_processor.get_elements(ds)
+    def get_tags_of_interest(self, dicom_file, tags_of_interest):
+        keys, subject_list = tags_of_interest + [SUBJECT_ID, SEQUENCE_NAME], []
+        for _, row in DicomTagProcessor._get_dicom_df(dicom_file).iterrows():
```
Comment (lwjason) [inline toolbox/dicom.py:112]: `DicomTagProcessor._get_dicom_df` -> `self._get_dicom_df`

File: toolbox/dicom.py [hunk 2]
```diff
@@ -97,43 +107,75 @@ def get_elements(self, ds):
...
+    subject specific DICOM files.
+    :return: absolute DICOM file path that corresponds to the middle slice.
+    """
+    slice_ids_dict = {}
+    for dcm in os.listdir(dicom_folder):
+        slice_id = get_file_number(dcm)
+        slice_ids_dict[slice_id] = os.path.join(dicom_folder, dcm)
+    slice_ids = list(slice_ids_dict.keys())
+    slice_ids.sort()
+    return slice_ids_dict[slice_ids[len(slice_ids)//2]]
```
Comment (lwjason) [inline toolbox/dicom.py:159]: I remember sometimes this doesn't give you the right middle slice. But maybe we are fine here as we just want to pick a slice

Comment (johnccshen) [reply toolbox/dicom.py:159]: maybe natsort?

File: toolbox/dicom.py (hunk 1 above)
Comment (cepa995) [reply toolbox/dicom.py:112]: Since `_get_dicom_df `is static, no `self `argument is needed.

File: toolbox/dicom.py (hunk 2 above)
Comment (cepa995) [reply toolbox/dicom.py:159]: I think we are fine, but I will check @johnccshen suggestion.

File: toolbox/dicom.py (hunk 1 above)
Comment (lwjason) [reply toolbox/dicom.py:112]: self._get_dicom_df is more common to call the static class function though

**Pull request merged**